soroban-env-host/src/builtin_contracts/stellar_asset_contract/balance.rs

```diff
@@ -773,10 +773,20 @@ fn set_trustline_authorization(
             )),
         }?;
 
+        let is_authorized = tl.flags & (TrustLineFlags::AuthorizedFlag as u32) != 0;
+
         if authorize {
+            if is_authorized {
```

Comment:
Both statements could be simplified to `if is_authorized == authorize`, but up to you.

Reply:
ah good point. fixed.